src/AspNetCore/Src/managedexports.cxx

```diff
@@ -355,4 +355,17 @@ http_cancel_io(
     return pHttpContext->CancelIo();
 }
 
+EXTERN_C __MIDL_DECLSPEC_DLLEXPORT
+HRESULT
+http_response_set_header(
+    _In_ IHttpContext* pHttpContext,
+    _In_ PCSTR pszHeaderName,
+    _In_ PCSTR pszHeaderValue,
+    _In_ USHORT usHeaderValueLength,
+    _In_ BOOL  fReplace
+)
+{
+    return pHttpContext->GetResponse()->SetHeader( pszHeaderName, pszHeaderValue, usHeaderValueLength, fReplace );
```

Comment:
There's an indexed overload you can use for known headers so you don't have to serialize and pin their names.
https://msdn.microsoft.com/en-us/library/ms694295(v=vs.90).aspx

Reply:
Is http_response_set_header even needed? can't the managed code call SetHeader directly?

Reply:
The same for http_set_response_status_code

Reply:
The managed code can call SetHeader, but it still needs to do a pinvoke. Either way we need to access the native iis http context. At this point I am keen on not changing it.

Reply:
I think we will need to add two native methods to handle this.

Reply:
will you set known header and unknown header differently at managed layer? IIS SetHeader api does not distinguish known header and unknown header here.

Reply:
The known header API avoids having the managed layer serialize extra strings. As these are the most common headers, that's a lot of allocation, encoding, and pinning we can avoid.